app/src/main/java/at/sw21_tug/team_25/expirydates/MainActivity.kt

```diff
@@ -26,5 +39,48 @@ class MainActivity : AppCompatActivity() {
         )
         setupActionBarWithNavController(navController, appBarConfiguration)
         navView.setupWithNavController(navController)
+        navMenu = navView.menu
+    }
+
+    fun refreshCurrentFragment(){
+        val id = navController.currentDestination?.id
+
+        val home = navMenu.findItem(R.id.navigation_home)
+        if (home != null) {
+            home.title = getString(R.string.title_home)
+        }
+        val add = navMenu.findItem(R.id.navigation_add)
+        if (add != null) {
+            add.title = getString(R.string.title_add)
+        }
+        val list = navMenu.findItem(R.id.navigation_list)
+        if (list != null) {
+            list.title = getString(R.string.title_list)
+        }
+        updateTitle()
+
+        navController.popBackStack(id!!,true)
+        navController.navigate(id)
+    }
+
+    fun updateTitle() {
+        when (navController.currentDestination?.label) {
+            "Добавлять" -> navController.currentDestination?.label = getString(R.string.title_add)
```

Comment:
how about comparing against currentDestination?.id ? this would prevent any hardcoded strings

Reply:
Good idea, thank you.